grpcgcp/gcp_balancer.go

```diff
@@ -447,5 +493,29 @@ func (gb *gcpBalancer) UpdateSubConnState(sc balancer.SubConn, scs balancer.SubC
 	}
 }
 
+// refresh initiates a new SubConn for a specific subConnRef and starts connecting.
+// If the refresh is already initiated for the ref, then this is a no-op.
+func (gb *gcpBalancer) refresh(ref *subConnRef) {
+	if ref.refreshing {
```

Comment:
hmm.... I will still need a lock after compare-and-swap to make sure the code below is not executed concurrently. If I place the lock after compare-and-swap there is no guarantee nothing intercept the lock between compare-and-swap and the lock, so I had to place it above, but I want to avoid locking if it is already refreshing. We can sync up on this to clarify the things.